db/mongo/mongodb.go

```diff
@@ -56,10 +56,10 @@ func (m *mongoClient) SaveTask(task *models.Task) error {
 
 	opts := options.Update().SetUpsert(true)
 	if _, err := collection.UpdateOne(context.TODO(), filter, update, opts); err != nil {
-	if _, err := collection.InsertOne(context.TODO(), task); err != nil {
-		return errors.Wrap(err, "failed to save a task")
+		if _, err := collection.InsertOne(context.TODO(), task); err != nil {
```

Comment:
why are you using two functions Update and Insert here?

Reply:
shahzad bhai it filters for existing task by id if the id is present it will update otherwise it will generate error and the task should be inserted as you instructed in onboarding task "SaveTask (add and update)"

Reply:
Our logic should be in the Service layer. In DB Layer, we write code that does the CRUD Operations only.

Reply:
ok

Reply:
Shahzad bhai pls help me on this one as i go through your code both update and add task is implemented separately but i have to do them in one function

Reply:
please explore setUpsert option in mongodb. 
If you are still unable to understand then we can discuss this on a call.